hw/ip/usbdev/data/usbdev_testplan.hjson

```diff
@@ -18,12 +19,824 @@
+      stage: V1
+      tests: ["usbdev_invalid_sync_pattern"]
+    }
+    {
+      name: false_eop
+      desc: '''
+            Verify that the usbdev must be able to recognize false EOP signal and indicate error.
+
+            - Generate an EOP with a single bit SE0 while device is configured for 2 bit SE0 for EOP.
+            - Verify that the interrupt pin goes high indicating an error due to false eop.
+            '''
+      stage: V1
+      tests: ["usbdev_false_eop"]
+    }
+    {
+      name: setup_stage
+      desc: '''
+            Verify the SETUP stage of enumeration.
+
+            - Send a setup token with a valid setup packet containing the address(0), endpoint number(0)
+              followed by Data0 packet requesting information about device i.e. descriptors.
+            - Verify that the device ACK the setup packet and prepare for the data stage.
+            '''
+      stage: V1
+      tests: ["usbdev_setup_stage"]
+    }
+    {
+      name: in_data_stage
+      desc: '''
+            Verify the IN Transfer with Device Descriptor Data.
+
+            - Issue an IN token to request control data from the device.
+            - Verify that the device should respond with a DATA packet containing the information
+              required by host or a NAK packet if there is no data available.
+            '''
+      stage: V1
+      tests: ["usbdev_in_data_stage"]
+    }
+    {
+      name: out_data_stage
+      desc: '''
+            Verify the OUT Transfer with DEVICE Settings.
+
+            - Send an OUT token followed by a data packet containing control data to the device.
+            - Verify that the device should acknowledge the successful receipt of data by sending
+              an ACK packet or return a NAK if it is still processing previous data.
+            '''
+      stage: V1
+      tests: ["usbdev_out_data_stage"]
+    }
+    {
+      name: out_status_stage
+      desc: '''
+            Verify the OUT Transfer for Status Reporting.
+
+            - If the host used IN tokens during data stage then status stage is initiated with an OUT token.
+            - Send an OUT token followed by a zero-length DATA0 data packet to the device.
+            - Verify that the device should acknowledge the successful receipt of the status request by
+              sending an ACK packet or return NAK if it is still processing or STALL if an error occurred.
+            '''
+      stage: V1
+      tests: ["usbdev_out_status_stage"]
+    }
+    {
+      name: in_status_stage
+      desc: '''
+            Verify the IN Transfer for Status Reporting.
+
+            - If the host used OUT tokens during data stage then status stage must start with an IN Token.
+            - Verify that the device must respond with zero length Data0 packet for succesful completion
+              STALL for error and NAK for busy followed by ACK from Host.
+            '''
+      stage: V1
+      tests: ["usbdev_in_status_stage"]
+    }
```

Comment:
May I suggest that block level DV should focus upon packet-level communications and not concern itself - at least for now - with the stages of the Control Transfers, please?

Control Transfers and communications above the packet/transaction level have already been extensively verified in FPGA with real host controllers and in top-level simulation with the USBDPI model, so this is a lower priority.

Reply:
@alees24 May I remove these test cases? or i need to change the descriptions.

Reply:
I'd suggest changing those tests to "stage: V2" at least for now, please.

Reply:
@alees24 , can you change all test points except "smoke" to V2?
As you see in other blocks, (https://opentitan.org/dashboard/index.html#top)
V1 are only csr tests (autogen) and smoke.

Reply:
It's a fair point.
They should all be V2 with the exception `usbdev_smoke` which has not yet been fleshed out.

May I suggest the following as a starting point for the smoke test description, if you don't already have one in a separate PR/work?
```
    {
      name: smoke
      desc: '''
            Smoke test of simple packet transmission in both directions over the USB.

            - DV environment programs CSRs for suitable pin configuration, leaving the device
              unconnected by not driving the DP pullup.
            - Program CSRs to enable SETUP packet reception and OUT traffic on Endpoint Zero.
            - Present a buffer description in the AvBuffer, to receive the SETUP packet.
            - Enable the interface (assert DP pull up) to indicate device presence.
            - DV host-side detects device presence.
            - DV host-side transmits a SETUP packet to Address 0, Endpoint 0.
            - Check that the description of the SETUP packet appears in the RxFiFo, with the
              expected 'setup' indicator, buffer ID, and a packet length of 8 bytes.
            - Check that the SETUP packet has been received into the packet buffer memory.
            - DV host-side checks that ACK was received from usbdev in response to the SETUP packet.
            '''
      stage: V1
      tests: ["usbdev_smoke"]
    }
```
This is about the minimal traffic/effort to check that transmission and reception occurs over the USB in each direction, without worrying about buffer management, interrupt handling, setting of device address etc.
HTH.